### PR TITLE
refactor(core): Clean up Prometheus service (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
Preparation for [PAY-1462](https://linear.app/n8n/issue/PAY-1462)

Follow-up to https://github.com/n8n-io/n8n/pull/10053 and https://github.com/n8n-io/n8n/pull/10050

### CODE REVIEW

File: packages/cli/src/metrics/prometheus-metrics.service.ts [hunk 1]
```diff
@@ -5,196 +5,200 @@ import promBundle from 'express-prom-bundle';
...
+			const labels = this.toLabels(event);
+
 			const counter = new promClient.Counter({
 				name: metricName,
-				help: `Total number of ${event.eventName} events.`,
-				labelNames: Object.keys(this.getLabelsForEvent(event)),
+				help: `Total number of ${eventName} events.`,
+				labelNames: Object.keys(labels),
 			});
+			counter.labels(labels);
```
Comment (ivov) [inline packages/cli/src/metrics/prometheus-metrics.service.ts:151]: There was a bug here - we were not using the labels other than for label names.

File: packages/cli/src/metrics/prometheus-metrics.service.ts [hunk 2]
```diff
@@ -5,196 +5,200 @@ import promBundle from 'express-prom-bundle';
...
+			autoregister: false,
+			includeUp: false,
+			includePath: this.includes.labels.apiPath,
+			includeMethod: this.includes.labels.apiMethod,
+			includeStatusCode: this.includes.labels.apiStatusCode,
+		});
 
-			app.use(['/rest/', '/webhook/', 'webhook-test/', '/api/'], metricsMiddleware);
-		}
+		app.use(['/rest/', '/webhook/', '/webhook-test/', '/api/'], metricsMiddleware);
```
Comment (ivov) [inline packages/cli/src/metrics/prometheus-metrics.service.ts:96]: Others to include? e.g. `/webhook-waiting/`, `/form-waiting` Or excluded on purpose?

Comment (csuermann) [reply packages/cli/src/metrics/prometheus-metrics.service.ts:96]: good idea to include these 👍

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.52.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.52.0)

**Pull request merged**